app/controllers/pieces_controller.rb

```diff
@@ -8,9 +8,25 @@ def show
   end
 
   def update
+    @piece = Piece.find(params[:id])
     x = piece_params[:x_position].to_i
     y = piece_params[:y_position].to_i
-    Piece.find_by_id(params[:id]).move_to!(x, y)
+
+    if @piece.puts_game_in_check?(x, y)
+      if current_user.id == @piece.game.p2_id && @piece.game.state == 'White King in Check'
+        @piece.move_to!(x, y)
+        flash.alert = 'White King is in Check.'
+      elsif current_user.id == @piece.game.p1_id && @piece.game.state == 'Black King in Check'
+        @piece.move_to!(x, y)
+        flash.alert = 'Black King is in Check.'
+      elsif current_user.id == @piece.game.p1_id && @piece.game.state == 'White King in Check' ||
+            current_user.id == @piece.game.p2_id && @piece.game.state == 'Black King in Check'
+        flash.alert = 'You cannot put yourself in Check.'
```

Comment:
I feel like this would be better placed in either a before action or something checked by valid_move? so it does not get all they way to the update action before checking.

Reply:
Good point. I will abstract it when I get home.